lib/elixir/test/elixir/kernel/with_test.exs

```diff
@@ -40,6 +40,10 @@ defmodule Kernel.WithTest do
     assert result == :error
   end
 
+  test "else conditions" do
+    assert with({:ok, res} <- 41, do: res, else: ({:error, error} -> error; res -> res + 1)) == 42
```

Comment:
There is a warning happening:  "warning: this clause cannot match because a previous clause at line 44 always matches".
Try to compile without assertion.
I think it may have something to do with expansion.